ZelBack/src/services/serviceHelper.js

```diff
@@ -156,29 +163,55 @@ function isDecimalLimit(value, decimals = 8) {
 }
 
 /**
- * To handle timeouts on axios connection.
- * @param {string} url URL.
- * @param {object} options Options object.
- * @returns {object} Response.
+ * A central place for axios get requests. The defaults are set in apiServer.js
+ * @param {string} url The request URL
+ * @param {*} userOptions Standard axios options
+ * @returns
  */
-// helper function for timeout on axios connection
-const axiosGet = (url, options = {}) => {
-  if (!options.timeout) {
-    // eslint-disable-next-line no-param-reassign
-    options.timeout = 20000;
-  }
-  const abort = axios.CancelToken.source();
-  const id = setTimeout(
-    () => abort.cancel(`Timeout of ${options.timeout}ms.`),
-    options.timeout,
-  );
-  return axios
-    .get(url, { cancelToken: abort.token, ...options })
-    .then((res) => {
-      clearTimeout(id);
-      return res;
-    });
-};
+async function axiosGet(url, userOptions = {}) {
+  const options = { ...userOptions };
+
+  if (!options.signal) options.signal = shc.signal;
+
+  return axios.get(url, options);
```

Comment:
the cancel token is not needed? I remember we added it as sometimes it did not stop after the 20 seconds but maybe axios fixed that already?

Reply:
I am supposing it might be done through the abort controller but reading through it I can't figure it out from it.

Reply:
It's not done through the abortController - that is more of a future thing, where we might want to reload the system gracefully.

There is a lot of incorrect info about axios on the internet, and the docs arern't that great, especially the "cancellation" doc here https://axios-http.com/docs/cancellation. I think it's very poorly worded, to the point of being misleading.

During testing, I wasn't able to get it to hang. I'm pretty sure (99%) that the timeout cancels all requests, including connection related issues, after the timeout. I think the docs are alluding that it will only cancel after the request timeout, if you want to cancel earlier, you need to use the abortController.

For example:

```javascript
const axios = require('axios').default

async function main() {
  await axios.get('http://3.4.5.6', { timeout: 5_000 }).catch(err => console.log(err.code))
}

main()
```

Will return `timeout of 5000ms exceeded` even though, obviously, it doesn't connect as the ip is not listening. 

I think the problem may have been that the timeout was set to 0, where it would hang indefinetely if you didn't have a way to cancel the request. However we are setting a fallback timeout globally now.

If you want, I can add a setTimeout with the abort controller, but I don't think it actually does anything. I probably need to read the Axios code and see what they're actually doing.

Reply:
Cool as, happy to add a timeout back in if we need it

yes - in my other pull that I'm still working through testing, I have updated axios to the latest version. Have tested in on node 14, works well